### PR TITLE
Introduce GetAvatarState for divided avatar state.

### PR DESCRIPTION
**Pull request by ipdae**
- support get divided avatar information

### CODE REVIEW

File: NineChronicles.Headless/GraphTypes/States/AgentStateType.cs [hunk 1]
```diff
@@ -29,9 +30,13 @@ public AgentStateType()
...
                     {
-                        if (context.Source.accountStateGetter(kv.Value) is { } state)
+                        try
                         {
-                            avatarStates.Add(new AvatarState((Dictionary)state));
+                            avatarStates.Add(context.Source.accountStateGetter.GetAvatarState(kv.Value));
+                        }
+                        catch (InvalidAddressException)
+                        {
+                            // pass
```
Comment (longfin) [inline NineChronicles.Headless/GraphTypes/States/AgentStateType.cs:39]: What we got here is that the addresses in `agentState.avatarAddresses` are not in the actual chain state. is it possible?

File: NineChronicles.Headless/AccountStateGetterExtension.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            if (accountStateGetter(avatarAddress) is Dictionary dictionary)
+            {
+                string[] keys =
+                {
+                    LegacyInventoryKey,
+                    LegacyWorldInformationKey,
+                    LegacyQuestListKey,
+                };
+
+                bool backward = false;
```
Comment (longfin) [inline NineChronicles.Headless/AccountStateGetterExtension.cs:23]: Is this a flag representing the avatar before the states were diveded?

Comment (ipdae) [reply NineChronicles.Headless/AccountStateGetterExtension.cs:23]: yes

File: NineChronicles.Headless/GraphTypes/States/AgentStateType.cs (hunk 1 above)
Comment (ipdae) [reply NineChronicles.Headless/GraphTypes/States/AgentStateType.cs:39]: no that's impossible. i'll delete this

File: NineChronicles.Headless/AccountStateGetterExtension.cs (hunk 2 above)
Comment (longfin) [reply NineChronicles.Headless/AccountStateGetterExtension.cs:23]: `backward` isn't direct naming for that IMO, Is there any other idiom for the (previously) unified avatar state? (e.g. v1?)

**Review by longfin — APPROVED** [review]

**Pull request merged**